std/algorithm/comparison.d

```diff
@@ -1911,3 +1915,101 @@ bool isPermutation(alias pred = "a == b", Range1, Range2)
         [mytuple(2, 3), mytuple(1, 2)]
     ));
 }
+
+enum ifTestable(T, alias pred = a => a) = __traits(compiles, { if ((pred(T()))) {} }) || isArray!T;
```

Comment:
Not sure what's going on here, but instead of `allSatisfy` you may want to use simple recursion: `either` is defined with `n` arguments iff `either` with the `n - 1` tail arguments compiles. We use this trick in a number of places in phobos.